context.go

```diff
@@ -243,6 +244,36 @@ func (w *responseWrapper[Res]) ResponseTrailer() http.Header {
 	return w.response.Trailer()
 }
 
+// errorResponseWrapper wraps an Error to a responseSource to propagate metadata.
+//
+// Error metadata is only exposed in trailers, not headers.
+type errorResponseWrapper struct {
+	base responseSource
+	err  *Error
+}
+
+func (w *errorResponseWrapper) ResponseHeader() http.Header {
+	if w.base != nil {
+		return w.base.ResponseHeader()
+	}
+	return make(http.Header)
+}
+
+func (w *errorResponseWrapper) ResponseTrailer() http.Header {
+	combined := make(http.Header)
+	if w.base != nil {
+		for k, v := range w.base.ResponseTrailer() {
+			combined[k] = v
+		}
+	}
+	if w.err != nil {
+		for k, v := range w.err.Meta() {
+			combined[k] = v
+		}
+	}
+	return combined
```

Comment:
This doesn't seem right. On the client side, we already get the response headers and trailers from the underlying streaming conn and then use those to populate `err.Meta()`. Won't this duplicate metadata? Also, `err.Meta()` is not able to distinguish between headers and trailers, while the underlying connection is, so if this does _not_ duplicate metadata, I think we have a problem that headers sent by the server will get mashed together in the error metadata and then incorrectly reported by the client as trailers.